src/components/DataBar/DataBar.tsx

```diff
@@ -0,0 +1,197 @@
+   */
+  text: string;
+  /**
+   * The value of the individual segment that accumulates towards the task goal.
+   */
+  value: number;
+  /**
+   * Color variant of the individual segment.
+   */
+  variant?: Variants;
+};
+
+export type Props = {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Max value to be represented by the data bar.
+   */
+  max: number;
+  /**
+   * A list of segments to be represented by the data bar.
+   */
+  segments: DataBarSegmentProps[];
+  /**
+   * Color variant of the data bar. Decorates the segments.
+   */
+  variant?: Variants;
+} & React.HTMLAttributes<HTMLElement>;
+
+/**
+ * ```ts
+ * import {DataBar} from "@chanzuckerberg/eds";
+ * ```
+ *
+ * A data bar component that represents the relative completion of a task represented by one or more segments.
+ *
+ * Example usage:
+ *
+ * ```tsx
+ * <DataBar
+ *  id="data-bar"
+ *  max={100}
+ *  segments={[
+ *    { value: 25, text: 'Segment 1' },
+ *    { value: 10, text: 'Segment 2' },
+ *    { value: 15, text: 'Segment 3' },
+ *  ]} />
+ * ```
+ */
+export const DataBar = ({
+  className,
+  max = 100,
+  segments,
+  variant = 'brand',
+  ...other
+}: Props) => {
+  /**
+   * Warns the developer if the data bar has no accessible name.
+   */
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    !other['id'] &&
+    !other['aria-label'] &&
+    !other['aria-labelledby']
+  ) {
+    console.warn('You must provide an accessible name for the data bar');
+  }
+
+  /**
+   * Calculates the total of the segment values.
+   */
+  const totalSegmentValue = segments.reduce(
+    (previousValue, segment) => previousValue + segment.value,
+    0,
+  );
+
+  const isFull = totalSegmentValue >= max;
+
+  /**
+   * Calculates the total width the segments will take up where the width is at least 5% of the max value.
+   */
+  let totalSegmentWidth = 0;
+  for (
+    let index = 0;
+    index < segments.length && totalSegmentWidth < max;
+    index++
+  ) {
+    const segment = segments[index];
+    const value = Math.max(segment.value, max / 20);
+    totalSegmentWidth += value;
+  }
+
+  const segmentComponents: React.ReactElement<typeof DataBarSegment>[] = [];
+
+  /**
+   * Manages focus of the progress bar elements and implements left and right arrow key interaction.
+   * 1) Used to set tabIndex of the segment components.
+   * 2) Collects a list of element refs to set focus.
+   * 3) Manage tabIndex to allow focus before setting focus.
+   * 4) Set focus on the next or previous element respective to the arrow key interaction.
+   */
+  /* 1 */
+  const [focusableElementIndex, setFocusableElementIndex] = React.useState(0);
+  const segmentsRef = React.useRef([]); /* 2 */
+  const handleOnKeyDown = (
+    e: React.KeyboardEvent<HTMLElement>,
+    index: number,
+  ) => {
+    if (e.key === 'ArrowLeft' && index > 0) {
+      setFocusableElementIndex(focusableElementIndex - 1); /* 3 */
```

Comment:
super nit: for consistency, could we use `index` throughout this function?

Reply:
Not sure I'm following, can you explain where we would be using `index` in the function other than where it's used currently?

Reply:
oh yeah, I was thinking:
```
setFocusableElementIndex(index - 1);
segmentsRef.current[index - 1].focus();
```

for some reason switching between `focusableElementIndex` and `index`  was confusing to me

Reply:
Ah sounds good, will change